src/borkdude/deps.clj

```diff
@@ -645,24 +643,20 @@ public class ClojureToolsDownloader {
       (.getPath (io/file *dir* ".cpcache"))
       user-cache-dir)))
 
-(defn calculate-config-paths [opts]
-  (let [deps-edn (calculate-deps-edn opts)
-        config-dir (calculate-config-dir)
-        install-dir (calculate-tools-dir)]
-    (if (:repro opts)
-      (if install-dir
-        [(.getPath (io/file install-dir "deps.edn")) deps-edn]
-        [deps-edn])
-      (if install-dir
-        [(.getPath (io/file install-dir "deps.edn"))
-         (.getPath (io/file config-dir "deps.edn"))
-         deps-edn]
-        [(.getPath (io/file config-dir "deps.edn"))
-         deps-edn]))))
-
-(defn- calculate-checksum [opts]
-  (let [config-paths (calculate-config-paths opts)
-        val*
+(defn calculate-config-paths [opts deps-edn config-dir install-dir]
```

Comment:
If we're exposing this, I'd rather pass in a map as extra arguments then won't break previous callers.
Perhaps we can do everything via the `opts` or maybe `{:opts opts :deps-edn ... :config-dir ..}` make more sense?

Reply:
But in this case, there are no previous callers, right? Prior to this PR this was all calculated inline.

Reply:
I know, but if we add another argument later, it _will_ break the callers of this arity, unless we add another arity.
Another reason to pass a map is that the order of 4 arguments is already pretty hard to remember, if we would add more later, even more so.

Reply:
Oh, I see what you mean. So, thinking about what this means for the API, as things are at the moment, my use case would look like:
```clojure
(let [opts (deps/parse-args command-line-args)
      deps-edn (deps/calculate-deps-edn opts)
      install-dir (deps/calculate-tools-dir)
      config-dir (deps/calculate-config-dir)
      config-paths (deps/calculate-config-paths opts deps-edn config-dir install-dir)]
  (deps/calculate-checksum opts config-paths))
```
With a change where we pass things around in opts, it would look like:
```clojure
(let [opts (deps/parse-args command-line-args)
      deps-edn (deps/calculate-deps-edn opts)
      install-dir (deps/calculate-tools-dir)
      config-dir (deps/calculate-config-dir)
      opts (assoc opts :deps-edn deps-edn :config-dir config-dir :install-dir install-dir
      config-paths (deps/calculate-config-paths opts)
      opts (assoc opts :config-paths config-paths)]
  (deps/calculate-checksum opts))
```
IMO this is starting to place a pretty heavy mental load on the caller, they have to figure out which keys are required in the map for each step of the calculation, and how to calculate those values - it's not terrible, but it's not trivial either and feels brittle.

For reference, if we allow the recalculation as in the earlier patch, the API looks like this:
```clojure
(let [opts (deps/parse-args command-line-args)]
  (deps/calculate-checksum opts))
```
Which is much simpler than either and has no positional args, at the cost of the redundant calculation. Personally, I think it's worth it, but it's a matter of taste really.

If we're sure we want something like that, we could have functions that augment opts, like:
```clojure
(let [opts (deps/parse-args command-line-args)
      opts (deps/with-deps-edn opts)
      opts (deps/with-tools-dir opts)
      opts (deps/with-config-dir opts)
      opts (deps/with-config-paths opts)]
  (deps/calculate-checksum opts))
```

Which could be rewritten as:
```clojure
(let [opts (-> (deps/parse-args command-line-args)
               (deps/with-deps-edn)
               (deps/with-tools-dir)
               (deps/with-config-dir)
               (deps/with-config-paths))]
  (deps/calculate-checksum opts))
```
Which also conveniently resolves the naming issue around the "calculate-" vars.